.expeditor/scripts/finish_release/publish_to_chocolatey.ps1

```diff
@@ -37,13 +37,20 @@ Write-Host "--- Publishing package"
 
 $valid_build_creator="Chef Expeditor"
 
-if($env:BUILDKITE_BUILD_CREATOR -eq $valid_build_creator) {
-  choco pack $tempDir/habitat.nuspec --version $Version
-  choco push habitat.$Version.nupkg -k $env:CHOCO_API_KEY --timeout 600
-} else {
-  Write-Host "--- NOT PUBLISHING: Build triggered by $env:BUILDKITE_BUILD_CREATOR and not $valid_build_creator"
-  Write-Host "choco pack $tempDir/habitat.nuspec --version $Version"
-  Write-Host "choco push habitat.$Version.nupkg -k <elided ChocoApiKey> --timeout 600"
+$pack_cmd = "choco pack $tempDir/habitat.nuspec --version $Version"
+$publish_cmd = "choco push habitat.$Version.nupkg --timeout 600"
+
+try {
+    if($env:BUILDKITE_BUILD_CREATOR -eq $valid_build_creator) {
+        Invoke-Expression $pack_cmd
```

Comment:
I don't think the error behavior of Invoke-Expression is what you expect;
If you run 
```
$test_fail = "test 1 -ne 1"
try {
    Invoke-Expression $test_fail
    if (-not $?) { throw "bad command" }
    Write-Host "Invoke succeeded"
} finally {
    Write-Host "Finally"
}
```
You get 
```
Invoke succeeded
Finally
```
It is apparently not like bash, where: 
```
% test 1 -ne 1; echo $?
1
```

Reply:
https://stackoverflow.com/questions/32348794/how-to-get-status-of-invoke-expression-successful-or-failed

Reply:
@markan is correct. What you want here is `if($LASTEXITCODE -ne 0)`. While `$?` will always evaluate to `$true` here, The choco exit code will persist in the `$LASTEXITCODE` variable.